bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/handler/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,410 @@
+ *
+ * @author Pauli Anttila, Yannick Schaus - Initial contribution
+ * @author Michael Lobstein - Improvements for OH3
+ */
+@NonNullByDefault
+public class EpsonProjectorHandler extends BaseThingHandler {
+    private static final int DEFAULT_POLLING_INTERVAL_SEC = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorHandler.class);
+
+    private EpsonProjectorDevice device = new EpsonProjectorDevice();
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private final SerialPortManager serialPortManager;
+
+    private boolean isPowerOn = false;
+
+    public EpsonProjectorHandler(Thing thing, SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+            EpsonProjectorConfiguration config = getConfigAs(EpsonProjectorConfiguration.class);
+            final String serialPort = config.serialPort;
+            final String host = config.host;
+            final Integer port = config.port;
+            final Integer pollingInterval = config.pollingInterval;
+
+            if (serialPort != null && !serialPort.equals("")) {
+                device = new EpsonProjectorDevice(serialPortManager, serialPort, scheduler);
+            } else if (host != null && !host.equals("") && port > 0 && port < 65535) {
+                device = new EpsonProjectorDevice(host, port, scheduler);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                return;
+            }
+
+            try {
+                updateStatus(ThingStatus.OFFLINE);
+                device.connect();
+
+                List<Channel> channels = this.thing.getChannels();
+
+                pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                    for (Channel channel : channels) {
+                        // only query power & lamp time when projector is off
+                        if (isPowerOn || (channel.getUID().getId().equals(CHANNEL_TYPE_POWER)
+                                || channel.getUID().getId().equals(CHANNEL_TYPE_LAMPTIME))) {
+                            updateChannelState(channel);
+                        }
+                    }
+                }, 0, (pollingInterval > 0) ? pollingInterval : DEFAULT_POLLING_INTERVAL_SEC, TimeUnit.SECONDS);
```

Comment:
You should avoid assigning class fields asynchronously since this handler might get disposed before you have a chance to assign it. This might lead to the case where this period task is never canceled.

Instead you should create this task during `initialize` and then inside the task just check if the device is connected or not before trying to update the channels.

Reply:
This should be fixed now.